### PR TITLE
[DDW-427] Handle new transaction fee calculation error messages

### PR DESCRIPTION
**Pull request by nikolaglumac**
This PR introduces more granular error messages for transaction-fee-calculation Api endpoint:
```
Amount + fees exceeds walletBalance:
- error.diagnostic.details.msg === 'Not enough coins to cover fee.'
= show "Not enough Ada for fees. Try sending a smaller amount."
-----
Amount exceeds availableBalance due to pending transactions:
- error.diagnostic.details.msg === 'Not enough available coins to proceed.'
- total walletBalance > error.diagnostic.details.availableBalance
= show "Can not calculate fees while there are pending transactions."
-----
Amount exceeds walletBalance:
- error.diagnostic.details.msg === 'Not enough available coins to proceed.'
- total walletBalance === error.diagnostic.details.availableBalance
= show "Not enough Ada. Try sending a smaller amount."
```

## Todo:

- [x] Bump Cardano version in `cardano-sl-src.json` to the commit which contains the fixed Api error response (https://github.com/input-output-hk/cardano-sl/pull/3710)
- [x] Inject missing Japanese translations:
```
"api.errors.CanNotCalculateTransactionFeesError": "Can not calculate fees while there are pending transactions.",
"api.errors.NotEnoughFundsForTransactionError": "Not enough Ada. Try sending a smaller amount.",
```

## Screenshots:

screen shot 2018-10-10 at 09 27 17
screen shot 2018-10-10 at 09 27 24
screen shot 2018-10-10 at 09 27 07

---

### Basics

- [ ] PR is updated to the most recent version of target branch (and there are no conflicts)
- [ ] PR has good description that summarizes all changes and shows some screenshots or animated GIFs of important UI changes
- [ ] CHANGELOG entry has been added and is linked to the correct PR on GitHub
- [ ] Automated tests: All acceptance tests are passing (`yarn run test`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`yarn run dev`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`yarn run package` / CI builds)
- [ ] There are no *flow* errors or warnings (`yarn run flow:test`)
- [ ] There are no *lint* errors or warnings (`yarn run lint`)
- [ ] Text changes are proofread and approved (Jane Wild)
- [ ] There are no missing translations (running `yarn run manage:translations` produces no changes)
- [ ] UI changes look good in all themes (Alexander Rukin)
- [ ] Storybook works and no stories are broken (`yarn run storybook`)
- [ ] In case of dependency changes `yarn.lock` file is updated

### Code Quality
- [ ] Important parts of the code are properly documented and commented
- [ ] Code is properly typed with flow
- [ ] React components are split-up enough to avoid unnecessary re-rendering
- [ ] Any code that only works in Electron is neatly separated from components

### Testing
- [ ] New feature / change is covered by acceptance tests
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature / change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review:
- [ ] Merge PR
- [ ] Delete source branch
- [ ] Move ticket to `done` on the Youtrack board

### CODE REVIEW

File: source/renderer/app/api/api.js [hunk 1]
```diff
@@ -352,7 +359,34 @@ export default class AdaApi {
...
+            // Amount exceeds walletBalance:
+            // - error.diagnostic.details.msg === 'Not enough available coins to proceed.'
+            // - total walletBalance === error.diagnostic.details.availableBalance
+            // = show "Not enough Ada. Try sending a smaller amount."
+            throw new NotEnoughFundsForTransactionError();
+          }
+        } else {
+          // Amount exceeds walletBalance:
+          // = show "Not enough Ada. Try sending a smaller amount."
+          throw new NotEnoughFundsForTransactionError();
```
Comment (KtorZ) [inline source/renderer/app/api/api.js]: _In theory_, this case shouldn't ever happen :no_mouth:

File: source/renderer/app/api/api.js [hunk 2]
```diff
@@ -352,7 +359,34 @@ export default class AdaApi {
...
+          throw new NotEnoughFundsForTransactionFeesError();
+        } else if (errorMessage.includes('Not enough available coins to proceed')) {
+          const availableBalance = new BigNumber(
+            get(error, 'diagnostic.details.availableBalance', 0)
+          ).dividedBy(LOVELACES_PER_ADA);
+          if (walletBalance.gt(availableBalance)) {
+            // Amount exceeds availableBalance due to pending transactions:
+            // - error.diagnostic.details.msg === 'Not enough available coins to proceed.'
+            // - total walletBalance > error.diagnostic.details.availableBalance
+            // = show "Cannot calculate fees while there are pending transactions."
```
Comment (KtorZ) [inline source/renderer/app/api/api.js]: Hmmm. Yes and no. I wonder if we couldn't teach users a bit here. It's not only about the fee, the funds just aren't available for making a transaction, regardless of the fee computation. 

Yet, if you think that it is better like it is from a UX perspective, then fine :)

File: source/renderer/app/i18n/locales/ja-JP.json [hunk 3]
```diff
@@ -3,11 +3,13 @@
   "ImageUploadWidget.dropFileHint": "ファイルをここにドロップしてください",
   "api.errors.AllFundsAlreadyAtReceiverAddressError": "送ろうとしている資金は既に送金先のアドレスにて反映済みです。",
   "api.errors.ApiMethodNotYetImplementedError": "このAPIは未実装です。",
+  "api.errors.CanNotCalculateTransactionFeesError": "処理中の取引がある間は、手数料を計算することができません。",
   "api.errors.ForbiddenMnemonicError": "ウォレットの復元パスフレーズが無効です。送信された復元パスフレーズは、ドキュメントの復元パスフレーズの一つですので、資金を保持するウォレットには使用しないでください。",
   "api.errors.GenericApiError": "エラーが発生しました。",
   "api.errors.IncorrectPasswordError": "無効なウォレットパスワード",
   "api.errors.NotAllowedToSendMoneyToRedeemAddressError": "Ada還元アドレスに送金することはできません。",
   "api.errors.NotAllowedToSendMoneyToSameAddressError": "送金元と入金先が同じアドレスである場合にはトランザクションは行えません。このウォレットに十分なアドレスがあることを確認するか、別のアドレスに送金を行なってください。",
+  "api.errors.NotEnoughFundsForTransactionError": "ADAが足りません。より少額の送金を試みてください。",
```
Comment (KtorZ) [inline source/renderer/app/i18n/locales/ja-JP.json]: Indeed :no_mouth:

File: source/renderer/app/api/api.js (hunk 2 above)
Comment (nikolaglumac) [reply source/renderer/app/api/api.js]: @KtorZ these error messages are displayed during transaction fee calculation so I think it is best to format them this way. The point is that you can not create a transaction without having transaction fees being calculated first. But I get your point too...

**Pull request merged**